auditbeat/module/auditd/audit_linux.go

```diff
@@ -454,3 +464,80 @@ func kernelVersion() (major, minor int, full string, err error) {
 
 	return major, minor, release, nil
 }
+
+func determineSocketType(c *Config, log *logp.Logger) (string, error) {
+	client, err := libaudit.NewAuditClient(nil)
+	if err != nil {
+		if c.SocketType == "" {
+			return "", errors.Wrap(err, "failed to create audit client")
+		}
+		// Ignore errors if a socket type has been specified. It will fail during
+		// further setup and its necessary for unit tests to pass
+		return c.SocketType, nil
+	}
+	defer client.Close()
+	status, err := client.GetStatus()
+	if err != nil {
+		if c.SocketType == "" {
+			return "", errors.Wrap(err, "failed to get audit status")
+		}
+		return c.SocketType, nil
+	}
+	rules, _ := c.rules()
+
+	isLocked := status.Enabled == auditLocked
+	hasMulticast := hasMulticastSupport()
+	hasRules := len(rules) > 0
+
+	const useAutodetect = "Remove the socket_type option to have auditbeat " +
```

Comment:
I like the logic change here. It's deterministic now if the user specifies a `socket_type`.